syntax = "proto3";
import "google/protobuf/empty.proto";

package agenda;

message Contact{
    string name = 1;
    string phone = 2;
}

message ContactsList {
    repeated Contact contacts = 1;
}

message Response {
    bool isSuccess = 1;
    string description = 2;
}

service AgendaService {
    rpc CheckConnection(google.protobuf.Empty) returns (Response) {};
    rpc GetAllContacts(google.protobuf.Empty) returns (ContactsList) {};
    rpc AddContact(Contact) returns (Response) {};
    rpc RemoveContact(Contact) returns (Response) {};
    rpc UpdateContact(Contact) returns (Response) {};
}